http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -199,6 +199,7 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
         s"Successfully created cluster ${dbCluster.getGkeClusterId.toString}! | trace id: ${ctx.traceId}"
       )
 
+      // TODO: Should we move nginx installation to a separate step so it doesn't fail cluster creation?
```

Comment:
I ran into a case where cluster was successfully created but nginx installation failed thereafter (because I'd specified an invalid nginx version). When I tried to create another app with the same project, it failed saying it can't create an app because the cluster is in `Error` status. It feels like we should **not** mark a cluster as `Error`ed just because ingress installation failed but rather incorporate checking if ingress is installed into the app creation flow. 

Any thoughts?

Reply:
Hmm... I think the cluster transitioning to `Error` status in this case is correct. The ingress is a cluster-level resource, created once at cluster creation time. I don't think each app should concern itself with whether cluster-wide resources are set-up properly.

Whether this should block future app creation is another issue though. Perhaps we could consider doing something like allowing the creation of one non-errored cluster per project, as opposed to a blanket enforcement of 1 per project like we do now?

Reply:
> The ingress is a cluster-level resource, created once at cluster creation time. I don't think each app should concern itself with whether cluster-wide resources are set-up properly.

That reasoning makes sense too.

I think we can recover from the case where we end up with an `Error`ed cluster and users can still create apps somehow, this issue would be sufficiently addressed. Your suggestion 

> we could also proactively clean-up at cluster creation time if an error occurs by introducing a status along the lines of 'DELETED_WITH_ERROR' to capture that 1) it is deleted in google and 2) an error occurred (I feel 2 is an important thing to keep a record of with the cluster and why just transitioning it to Deleted isn't sufficient)

sounds like a good solution to me 😊